lib/user.rb

```diff
@@ -0,0 +1,22 @@
+class User
+  attr_reader :name, :jokes, :told
+  def initialize(name)
+    @name = name
+    @jokes = []
+    @told = false
+  end
+
+  def learn(joke)
+    @jokes << joke
+  end
+
+  def tell(user, joke)
+    @user = user
```

Comment:
this would be necessary if we needed to track who user 2 was through user 1, like a "who was the last person I told a joke to", but we didn't ask for that kind of functionality